agents-common/src/main/java/org/apache/ranger/plugin/policyevaluator/RangerDefaultPolicyEvaluator.java

```diff
@@ -1512,4 +1504,27 @@ private List<RangerConditionEvaluator> createPolicyConditionEvaluators(RangerPol
 		return ret;
 	}
 
+	private static boolean isMatchForResourceMatchingScope(final RangerAccessRequest.ResourceMatchingScope scope, final RangerPolicyResourceMatcher.MatchType matchType, boolean isAnyMatch) {
+		boolean ret = false;
+		if (isAnyMatch){
+			ret = matchType !=  RangerPolicyResourceMatcher.MatchType.NONE;
+		}
+		else if (scope!=null) {
+				switch (scope) {
+						case SELF_OR_DESCENDANTS: {
+							ret = matchType != RangerPolicyResourceMatcher.MatchType.NONE;
+							break;
+						}
+						case SELF_AND_ALL_DESCENDANTS: {
+							ret = matchType != RangerPolicyResourceMatcher.MatchType.NONE;
```

Comment:
shouldn't matchType be SELF_AND_ALL_DESCENDENTS here?

Reply:
matchType=DESCENDANT is the scenario we are also interested in.
if policy resource is (t1,cf1,c1) and accessed resource is (t1,cf1) then matchType is DESCENDANT which is what we want to match as it is a valid policy to evaluate for this case.

Reply:
@fateh288  - case blocks for `SELF_OR_DESCENDANTS` and `SELF_AND_ALL_DESCENDANTS` are the same, which doesn't look correct. Can you review?

Reply:
Yes, 'case' blocks for both SELF_OR_DESCENDANTS and SELF_AND_ALL_DESCENDANTS would be the same. 

ResourceMatchingScope=SELF_OR_DESCENDANTS will result in isMatched = true for all cases of matchType in {SELF, DESCENDANT, ANCESTOR, SELF_AND_ALL_DESCENDANTS} --- I am not sure here if matchType is Ancestor then why isMatched=true makes sense. 
For ResourceMatchingScope=SELF_OR_DESCENDANTS, I tried changing 
`ret = matchType == RangerPolicyResourceMatcher.MatchType.SELF || matchType == RangerPolicyResourceMatcher.MatchType.SELF_AND_ALL_DESCENDANTS || matchType == RangerPolicyResourceMatcher.MatchType.DESCENDANT;`

After this change, agents-common test cases pass successfully even after I change it to above (do you suggest changing it to this ?).

The same works for SELF_AND_ALL_DESCENDANTS too.
 
In the current implementation, the difference between SELF_OR_DESCENDANTS and SELF_AND_ALL_DESCENDANTS comes in implementation of updateAccessResult() https://github.com/apache/ranger/blob/c9ee929d025dd73689ba1f242e3efdd23d8c9c8a/agents-common/src/main/java/org/apache/ranger/plugin/policyevaluator/RangerDefaultPolicyEvaluator.java#L533

Reply:
@mneethiraj @kulkabhay 
Any ideas on how to proceed? 
I have resolved the merge conflicts.
Essentially with SELF_AND_ALL_DESCENDANTS we want same behavior as SELF_OR_DESCENDANTS but the special case for matchType!=DESCENDANTS needs to be bypassed